test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,227 @@
+
+	ns := "namespace"
+
+	br, e := scClient.Brokers().Create(&v1alpha1.Broker{})
+	if nil == e {
+		t.Fatal("needs a name", br.Name)
+	}
+	sc, e := scClient.ServiceClasses(ns).Create(&v1alpha1.ServiceClass{})
+	if nil == e {
+		t.Fatal("needs a name", sc.Name)
+	}
+	i, e := scClient.Instances(ns).Create(&v1alpha1.Instance{})
+	if nil == e {
+		t.Fatal("needs a name", i.Name)
+	}
+	bi, e := scClient.Bindings(ns).Create(&v1alpha1.Binding{})
+	if nil == e {
+		t.Fatal("needs a name", bi.Name)
+	}
+}
+
+func TestBroker(t *testing.T) {
+	client, shutdownServer := getFreshApiserverAndClient(t)
+	defer shutdownServer()
+	brokerClient := client.Servicecatalog().Brokers()
+
+	broker := &v1alpha1.Broker{
+		ObjectMeta: v1.ObjectMeta{Name: "test-broker"},
+		Spec: v1alpha1.BrokerSpec{
+			URL:          "https://example.com",
+			AuthUsername: "auth username field value",
+			AuthPassword: "auth password field value",
+			OSBGUID:      "OSBGUID field",
+		},
+	}
+
+	// start from scratch
+	brokers, err := brokerClient.List(v1.ListOptions{})
+	if len(brokers.Items) > 0 {
+		t.Fatalf("brokers should not exist on start, had %v brokers", len(brokers.Items))
+	}
+
+	brokerServer, err := brokerClient.Create(broker)
+	if nil != err {
+		t.Fatal("error creating the broker", broker)
+	}
+	if broker.Name != brokerServer.Name {
+		t.Fatalf("didn't get the same broker back from the server \n%+v\n%+v", broker, brokerServer)
+	}
+
+	brokers, err = brokerClient.List(v1.ListOptions{})
+	if 1 != len(brokers.Items) {
+		t.Fatalf("should have exactly one broker, had %v brokers", len(brokers.Items))
+	}
+
+	brokerServer, err = brokerClient.Get(broker.Name)
+	if broker.Name != brokerServer.Name &&
+		broker.ResourceVersion == brokerServer.ResourceVersion {
+		t.Fatalf("didn't get the same broker back from the server \n%+v\n%+v", broker, brokerServer)
+	}
+
+	// check that the broker is the same both ways
+	brokerListed := &brokers.Items[0]
+	if !reflect.DeepEqual(brokerServer, brokerListed) {
+		t.Fatalf("didn't get the same broker twice", brokerServer, brokerListed)
+	}
+
+	brokerServer.Spec.AuthUsername = "dug"
+	brokerServer.Spec.AuthPassword = "paul"
+	brokerUpdated, err := brokerClient.Update(brokerServer)
+	if nil != err ||
+		"dug" != brokerUpdated.Spec.AuthUsername ||
+		"paul" != brokerUpdated.Spec.AuthPassword {
+		t.Fatal("broker wasn't updated", brokerServer, brokerUpdated)
+	}
+
+	brokerServer, err = brokerClient.Get("test-broker")
+	if nil != err ||
+		"dug" != brokerServer.Spec.AuthUsername ||
+		"paul" != brokerServer.Spec.AuthPassword {
+		t.Fatal("broker wasn't updated", brokerServer)
+	}
+
+	err = brokerClient.Delete("test-broker", &v1.DeleteOptions{})
+	if nil != err {
+		t.Fatal("broker should be deleted", err)
+	}
+
+	brokerDeleted, err := brokerClient.Get("test-broker")
+	if nil == err {
+		t.Fatal("broker should be deleted", brokerDeleted)
+	}
+}
+
+func getFreshApiserverAndClient(t *testing.T) (servicecatalogclient.Interface, func()) {
```

Comment:
> so it would could serve better to have a wipe etcd function

Aren't you going to need that anyway? It doesn't appear to be present now, nor does it appear that etcd restarts between test cases. That means you're currently starting each test in a dirty state.

> If we could get a randomly assigned port and...

This is making me think... although I am not, in principle, opposed in any way to you including etcd in our containerized development environment and running it there, I'm wondering what the chances are here of including etcd as a _library_ in glide.yaml instead and using it _programmatically_ to start embedded etcd servers here in-process, each on its own port, as you have suggested.

That is a non-blocking thought.